pkg/kubelet/cm/dra/manager.go

```diff
@@ -49,28 +46,20 @@ func NewManagerImpl(kubeClient clientset.Interface) (*ManagerImpl, error) {
 	klog.V(2).InfoS("Creating DRA manager")
 
 	manager := &ManagerImpl{
-		resources:  newClaimedResources(),
+		cache:      newClaimInfoCache(),
 		kubeClient: kubeClient,
 	}
 
 	return manager, nil
 }
 
 // Generate container annotations using CDI UpdateAnnotations API.
-func generateCDIAnnotation(
+func generateCDIAnnotations(
 	claimUID types.UID,
 	driverName string,
 	cdiDevices []string,
 ) ([]kubecontainer.Annotation, error) {
-	const maxKeyLen = 63 // max length of the CDI annotation key
-
-	deviceID := string(claimUID)
-
-	if len(deviceID) > maxKeyLen-len(driverName)-1 {
-		deviceID = deviceID[:maxKeyLen-len(driverName)-1]
-	}
-
-	annotations, err := cdi.UpdateAnnotations(map[string]string{}, driverName, deviceID, cdiDevices)
```

Comment:
My original comment on this:
> This appears to rely on internal knowledge of what cdi.UpdateAnnotations() does under the hood in terms of combining driverName with deviceID to create the annotation key. It seems like this functionality should be built into CDI somehow though, including returning an error if the resulting name does not have the right format (e.g. chopping off so much that the resulting string no longer is a DNS Subdomain name).

You seem to be simply removing this without explanation now though.

Is it not necessary at all? Or has then functionality been added into CDI inside the `UpdateAnnotations()` call, so it's no longer necessary?

Reply:
I removed it as all our tests pass without this code.
I'm going to ask CDI devs if it makes sense to implement the check in the cdi.UpdateAnnotations.

Reply:
OK. I know I ran into trouble here when I was originally trying to generate the CDI specs on the fly in the NVIDIA plugin. Because the annotations currently produced by this don't care how long they get and then trying to pass them through the CRI fails validation.

Reply:
Would failing cdi.UpdateAnnotations call if len(deviceID) > maxKeyLen-len(driverName)-1 be enough or you'd prefer cutting it somehow as the removed code did?

Reply:
It seems that cdi.UpdateAnnotation check is already in the code: https://github.com/container-orchestrated-devices/container-device-interface/blob/main/pkg/cdi/annotations.go#L101

From my point of view it's enough to at least get the issue noticed and reported. Do you think we should go further and cut deviceID somehow?

Reply:
Yeah. I think it's fine for now. Let's leave it, but stay aware of it.

Reply:
ok. I asked CDI developer to look at this.

Reply:
@klihub the original comment was about using internal cdi knowledge about maxKeyLen value to cut the deviceID if it's too long. Would it make sense to make it a constant accessible from outside of cdi?